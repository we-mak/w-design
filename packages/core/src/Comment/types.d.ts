import { ReactChild } from "react";
import { GlobProps } from "../typings";

export interface CommentProps extends GlobProps {
  message: string;
  time?: string;
  userName?: string | ReactChild;
  alt?: string;
  avatarSize?: "xs" | "sm" | "md" | "lg" | "xl" | "xxl";
  avatarUrl?: string;
  actions?: ReactChild | ReactChild[]; // Comment actions
  editEditor?: ReactChild; // edit comment Editor
  children?: any;
  style?: Object;
}
