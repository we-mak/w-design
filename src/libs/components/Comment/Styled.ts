import styled from "styled-components";

export const Message = styled.div`
  font-size: ${props => props.theme.fontSizes[1]};
  color: ${props => props.theme.colors["N70"]};
`;

export const CommentContainer = styled.div`
  width: 100%;
  margin: 0.8rem 0;
`;
CommentContainer.displayName = "CommentContainer";
