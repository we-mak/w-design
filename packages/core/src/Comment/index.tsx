import React, { FC } from "react";
import Avatar from "../Avatar";
import AvatarGroup from "../AvatarGroup";
import { Message, CommentContainer } from "./Styled";
import { CommentProps } from "./types";

const Comment: FC<CommentProps> = ({
  time,
  avatarSize,
  avatarUrl,
  userName,
  alt,
  message,
  editEditor,
  actions,
  children,
  ...rest
}) => {
  return (
    <CommentContainer {...rest}>
      <AvatarGroup
        avatar={
          <Avatar size={avatarSize} avatarUrl={avatarUrl} alt={alt}></Avatar>
        }
        name={userName}
        description={
          <>
            <time>{time}</time>
            {editEditor ? (
              editEditor
            ) : (
              <Message
                dangerouslySetInnerHTML={{
                  __html: message
                }}
              ></Message>
            )}
            {actions}
            {children}
          </>
        }
      ></AvatarGroup>
    </CommentContainer>
  );
};

export default Comment;
export { CommentProps };
