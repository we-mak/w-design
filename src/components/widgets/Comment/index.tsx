import React, { FC } from "react";
import Avatar from "../../../components/elements/Avatar";
import AvatarGroup from "../../../components/components/AvatarGroup";
import { CommentProps } from "./types";
import { Message, CommentContainer } from "./Styled";

const Comment: FC<CommentProps> = ({
  time,
  avatarSize,
  avatarUrl,
  userName,
  alt,
  message,
  editEditor,
  actions,
  children,
  ...rest
}) => {
  return (
    <CommentContainer {...rest}>
      <AvatarGroup
        avatar={<Avatar size={avatarSize} avatarUrl={avatarUrl} alt={alt}></Avatar>}
        name={userName}
        description={
          <>
            <time>{time}</time>
            {editEditor ? (
              editEditor
            ) : (
              <Message
                dangerouslySetInnerHTML={{
                  __html: message
                }}
              ></Message>
            )}
            {actions}
            {children}
          </>
        }
      ></AvatarGroup>
    </CommentContainer>
  );
};

export default Comment;
