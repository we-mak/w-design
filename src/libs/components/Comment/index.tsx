import React, { FC } from "react";
import Avatar from "../../elements/Avatar";
import AvatarGroup from "../AvatarGroup";
import { CommentProps } from "./types";
import { Message, CommentContainer } from "./Styled";

const Comment: FC<CommentProps> = ({
  time,
  avatarSize,
  avatarUrl,
  userName,
  alt,
  message,
  editEditor,
  actions,
  children,
  ...rest
}) => {
  return (
    <CommentContainer {...rest}>
      <AvatarGroup
        avatar={<Avatar size={avatarSize} avatarUrl={avatarUrl} alt={alt}></Avatar>}
        name={userName}
        description={
          <>
            <time>{time}</time>
            {editEditor ? (
              editEditor
            ) : (
              <Message
                dangerouslySetInnerHTML={{
                  __html: message
                }}
              ></Message>
            )}
            {actions}
            {children}
          </>
        }
      ></AvatarGroup>
    </CommentContainer>
  );
};

export default Comment;
