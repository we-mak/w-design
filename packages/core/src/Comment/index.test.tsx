import * as React from "react";
import { render, cleanup } from "@testing-library/react";
import { Provider } from "../..";
import Comment from "./index";

afterEach(cleanup);

describe("<Comment/>", () => {
  // Render test
  it("should renders correctly", () => {
    const { container } = render(
      <Provider>
        <Comment
          avatarSize="xs"
          avatarUrl="https://source.unsplash.com/random"
          userName=""
          alt=""
          time=""
          message=""
          actions={[]}
          editEditor={<div>...</div>}
        ></Comment>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
