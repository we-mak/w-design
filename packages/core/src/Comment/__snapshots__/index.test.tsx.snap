// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment/> should renders correctly 1`] = `
<div>
  <div
    class="sc-crrsfI ldOFCl"
  >
    <div
      class="sc-gKsewC gBbyGQ"
    >
      <div
        class="sc-iBPRYJ bkigGD"
      >
        <figure
          class="sc-bdfBwQ eZmttX"
          style="background-color: white;"
        >
          <span
            class="sc-dlfnbm kXdhLK"
          >
            Wm
          </span>
          <img
            alt=""
            class="sc-hKgILt kmlAc sc-jSgupP eAjhPD"
            data-src="https://source.unsplash.com/random"
            loading="lazy"
            src="data:image/gif;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mP88R8AAvUB+VkkrXoAAAAASUVORK5CYII="
          />
        </figure>
      </div>
      <div
        class="sc-fubCfw ffnjez"
      >
        
        <div
          class="sc-jrAGrp fpfFp"
        >
          <time />
          <div>
            ...
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
